### PR TITLE
Implement classes

### PR DESCRIPTION
**Pull request by thecodechaser**
## This branch includes the following features:
-  Created classes Person, Teacher, Student.
-  Created constructor methods for all classes.
-  Initialized instance variables for all classes.
-  Created setters and getters for Person class.
-  Created required methods.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_read :id, :name, :age
+  attr_write :name, :age
```
Comment (mikethreels) [inline person.rb]: - [ ] the keywords should be `attr_reader` & `attr_writer` please make sure to change this before requesting another code review.
- [ ] [OPTIONAL] you can also use the `attr_accessor` keyword to combine the read and write function.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(age, name, parent_permission, classroom)
```
Comment (mikethreels) [inline student.rb]: - [ ] in the initialize method the name and parent_permission arguments should have a default value.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(age, name, parent_permission, specialization)
```
Comment (mikethreels) [inline teacher.rb]: - [ ] in the initialize method the name and parent_permission arguments should have a default value.

File: README.md [hunk 4]
```diff
@@ -1 +1,79 @@
...
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Inspiration: Microverse
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (mikethreels) [inline README.md:79]: - [ ] the link to the license is not working, please make sure it is working as in [the template](https://github.com/microverseinc/readme-template) provided by Microverse.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ranjeet, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @thecodechaser, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**